pkg/controller/controller.go

```diff
@@ -129,20 +129,23 @@ func (c *Controller) Run() error {
 			switch event.Type {
 			case "ADDED":
 				clusterSpec := &event.Object.Spec
-				stopC := make(chan struct{})
-				c.stopChMap[clusterName] = stopC
-
-				nc := cluster.New(c.KubeCli, clusterName, c.Namespace, clusterSpec, stopC, &c.waitCluster)
-				c.clusters[clusterName] = nc
-				analytics.ClusterCreated()
 
 				backup := clusterSpec.Backup
 				if backup != nil && backup.MaxSnapshot != 0 {
 					err := k8sutil.CreateBackupReplicaSetAndService(c.KubeCli, clusterName, c.Namespace, *backup)
 					if err != nil {
-						panic(err)
+						log.Errorf("cluster (%s) is dead due to failure to create backup: %v", clusterName, err)
```

Comment:
Right. There are two points we need to make here: 1. fail to create backup; 2. cluster is dead as we think.